kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigMapWatcher.java

```diff
@@ -170,9 +172,24 @@ private void processConfigMapDeleted(ConfigMap configMap) {
             propertySource = KubernetesUtils.configMapAsPropertySource(configMap);
         }
         if (passesIncludesExcludesLabelsFilters(configMap)) {
-            this.environment.removePropertySource(propertySource);
             KubernetesConfigurationClient.removePropertySourceFromCache(propertySource.getName());
-            this.environment = environment.refresh();
+            refreshEnvironment();
+        }
+    }
+
+    /**
+     * Send a {@link RefreshEvent} when a {@link ConfigMap} change affects the {@link Environment}
+     *
+     * @see io.micronaut.management.endpoint.refresh.RefreshEndpoint#refresh(Boolean)
+     */
+    private void refreshEnvironment() {
+        final Map<String, Object> changes = environment.refreshAndDiff();
+        if (LOG.isTraceEnabled()) {
+            LOG.trace("Changes in ConfigMap property sources: [{}]", String.join(", ", changes.keySet()));
+        }
+        if (!changes.isEmpty()) {
+            LOG.trace("Sending RefreshEvent");
```

Comment:
Add this log line under conditional block or remove it. I prefer to remove it.

Reply:
Done.